src/EventListener/AddFormatListener.php

```diff
@@ -31,26 +31,28 @@
 final class AddFormatListener
 {
     private $negotiator;
+    private $resourceMetadataFactory;
     private $formats = [];
-    private $mimeTypes;
     private $formatsProvider;
     private $formatMatcher;
 
     /**
-     * @throws InvalidArgumentException
+     * @param ResourceMetadataFactoryInterface|FormatsProviderInterface|array $resourceMetadataFactory
      */
-    public function __construct(Negotiator $negotiator, /* FormatsProviderInterface */ $formatsProvider)
+    public function __construct(Negotiator $negotiator, $resourceMetadataFactory, array $formats = [])
```

Comment:
I think we need to do something like this in so many places. Otherwise we'll end up with lots of holes in our arguments in 3.0 :laughing: